### PR TITLE
Home

### PR DESCRIPTION
**Pull request by Iyanuoluwa-abolade**
- Completed the UI for my main page, sign in, and sign up page.
- Added the password view toggle button as a stretch feature.

Image 7-3-24 at 5 19 PM
Image 7-3-24 at 5 20 PM

Image 7-3-24 at 5 20 PM (1)
Image 7-3-24 at 5 20 PM

### CODE REVIEW

File: backend/index.js [hunk 1]
```diff
@@ -1,10 +1,86 @@
-const { PrismaClient } = require('@prisma/client')
-const prisma = new PrismaClient()
-const cors = require('cors')
-const express = require('express')
+// const { PrismaClient } = require('@prisma/client')
```
Comment (robintan98) [inline backend/index.js]: Let's remember to not include commented code in code commits -- totally fine to have them locally, but just remember to remove them when pushing a commit to GitHub

File: backend/index.js [hunk 2]
```diff
@@ -1,10 +1,86 @@
...
+app.use(
+    session({
+        secret: 'SECRET',
+        resave: false,
+        saveUninitialized: false,
+        store: sessionStore,
+        cookie: {
+            sameSite: 'false',
+            secure: false,
+            expires: new Date(Date.now() + 365 * 24 * 60 * 60 * 1000),
```
Comment (robintan98) [inline backend/index.js]: Let's put 365 * 24 * 60 * 60 * 1000 as a constant called YEAR_TO_MILLISECOND_CONVERSION_FACTOR

File: backend/routes/users.js [hunk 3]
```diff
@@ -1,60 +1,210 @@
+// import express from "express";
```
Comment (robintan98) [inline backend/routes/users.js]: Same here, let's remove commented out code

File: backend/routes/users.js [hunk 4]
```diff
@@ -1,60 +1,210 @@
...
+                    Password: hashedPassword,
+                    Role,
+                },
+            });
+            req.session.user = user;
+            res.json({ user });
+
+        } catch (err) {
+            console.error(err);
+            res.status(500).json({ message: err.message });
```
Comment (robintan98) [inline backend/routes/users.js:50]: Can we include better error handling? For instance: rather than wrapping everything in a try catch, perform each of the prisma/bcrypt action's one-by-one, try-catching each of them if necessary. This helps us debug more efficiently -- think how easy it would've been for us to debug a few days ago if we had more specific error handling

File: backend/server.js [hunk 5]
```diff
@@ -1,41 +1,49 @@
...
-            secure: false,
-            expires: new Date(Date.now() + (365 * 24 * 60 * 60 * 1000)) // 1 year in milliseconds
-        }
-    })
-);
-
-sessionStore.sync();
-
-app.use( userRoutes );
+// import express from 'express';
```
Comment (robintan98) [inline backend/server.js]: If we're commenting out all of this code, should we just remove the file?

File: frontend/my-adopthive-app/src/SignIn/SignIn.css [hunk 6]
```diff
@@ -3,21 +3,23 @@
...
 
 .sign-in {
-    background-color: #ccc;
     padding: 70px;
     border-radius: 5px;
-    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
     text-align: center;
     max-width: 400px;
     width: 100%;
+    /* background-color: #ccc;
```
Comment (robintan98) [inline frontend/my-adopthive-app/src/SignIn/SignIn.css]: Remove commented out code again. This applies to all instances so I'll trust you to remove them by the next version

File: backend/routes/users.js [hunk 7]
```diff
@@ -74,8 +11,13 @@ const saltRounds = 10;
 env.config();
 
 router.post("/signup", async (req, res) => {
-    const { FirstName, MiddleName, LastName, Username, Password, ConfirmPassword, Role } = req.body;
+    console.log(req.body)
```
Comment (robintan98) [inline backend/routes/users.js]: Let's remove the console.log. We shouldn't log things in production for fear of cluttering and exposing private information to a user's console.

**Review by robintan98 — APPROVED** [review]
Other than the commented out code, which should ideally be removed in previous commits, the changes look good to me!

**Review by robintan98 — CHANGES_REQUESTED** [review]
Please address all comments in this branch. Also, let's name the commits more descriptively. For instance, rather than "update backend", we can say "integrated the authentication service with the home page".

File: backend/routes/users.js [hunk 8]
```diff
@@ -1,60 +1,210 @@
...
+// export default router;
+
+
 import express from "express";
 import bcrypt from "bcrypt";
-import { user } from "../models/user.js";
-import { Op } from "sequelize";
+import env from "dotenv";
+import { PrismaClient } from "@prisma/client";
+// import { User } from "../models/user.js";
```
Comment (aliceyang121) [inline backend/routes/users.js]: Same here, remove unused imports

File: backend/routes/users.js (hunk 4 above)
Comment (Iyanuoluwa-abolade) [reply backend/routes/users.js:50]: Please, can you expatiate more on this. I don't really understand what you mean. Thank you.

File: backend/prisma/schema.prisma [hunk 9]
```diff
@@ -20,4 +20,10 @@ model user {
   LastName    String
   Username    String @unique
   Password    String
+  role        Role
```
Comment (markhopson) [inline backend/prisma/schema.prisma:23]: Assuming you want to capitalize "Role" to match other columns

File: frontend/my-adopthive-app/src/Home/Home.jsx [hunk 10]
```diff
@@ -1,22 +1,16 @@
...
-//         </div>
-//     );
-// }
-
-// export default Home;
-
-function Home () {
-    return <>Welcome</>
+function Home() {
+  const { user, updateUser } = useContext(UserContext);
```
Comment (markhopson) [inline frontend/my-adopthive-app/src/Home/Home.jsx]: `updateUser` seems unnecessary?

File: frontend/my-adopthive-app/src/index.css [hunk 11]
```diff
@@ -1,68 +1 @@
-/* :root {
-  font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
-  line-height: 1.5;
-  font-weight: 400;
```
Comment (markhopson) [inline frontend/my-adopthive-app/src/index.css]: Delete this file?

File: frontend/my-adopthive-app/src/UserContext.js [hunk 12]
```diff
@@ -2,20 +2,4 @@ import { createContext } from 'react';
 
 export const UserContext = createContext();
```
Comment (markhopson) [inline frontend/my-adopthive-app/src/UserContext.js:3]: Why is this file still needed?

**Review by markhopson — COMMENTED** [review]
Left some comments.

File: frontend/my-adopthive-app/src/UserContext.js (hunk 12 above)
Comment (Iyanuoluwa-abolade) [reply frontend/my-adopthive-app/src/UserContext.js:3]: It is needed for the authentication and if deleted there won't be a user authentication

File: backend/prisma/schema.prisma (hunk 9 above)
Comment (Iyanuoluwa-abolade) [reply backend/prisma/schema.prisma:23]: When I tried capitalizing this, it was getting mixed up with the enumerated type. Therefore, they have to be different

File: backend/index.js [hunk 13]
```diff
@@ -31,7 +31,7 @@ app.use(
 );
 app.use(
     session({
-        secret: 'SECRET',
+        secret: 'TOPSECRETWORD',
```
Comment (aliceyang121) [inline backend/index.js:34]: What is the purpose of this?

Comment (Iyanuoluwa-abolade) [reply backend/index.js:34]: It is a key for the session middleware

**Review by robintan98 — APPROVED** [review]

**Review by robintan98 — APPROVED** [review]

**Comment by robintan98** [comment]
Changes look good -- let's merge this into main branch and get started on the TCs. Next time, let's have more descriptive names of the commits. For instance, instead of saying "UI-Development-Milestones", say "remove port=3000 from users.js".

File: backend/server.js (hunk 5 above)
Comment (Iyanuoluwa-abolade) [reply backend/server.js]: I will remove it

File: frontend/my-adopthive-app/src/Home/Home.jsx (hunk 10 above)
Comment (Iyanuoluwa-abolade) [reply frontend/my-adopthive-app/src/Home/Home.jsx]: Noted
